plugins/modules/cloudwatch_metric_alarm_info.py

```diff
@@ -0,0 +1,203 @@
+            returned: always
+            type: str
+        ok_actions:
+            description: The actions to execute when this alarm transitions to an OK state from any other state.
+            returned: always
+            type: str
+        period:
+            description:
+                - The length, in seconds, used each time the metric specified in MetricName is evaluated.
+                - Valid values are 10, 30, and any multiple of 60.
+            returned: always
+            type: int
+        state_reason:
+            description: An explanation for the alarm state, in text format.
+            returned: always
+            type: str
+        state_reason_data:
+            description: An explanation for the alarm state, in JSON format.
+            returned: always
+            type: str
+        state_updated_timestamp:
+            description: The time stamp of the last update to the alarm state.
+            returned: always
+            type: str
+        state_value:
+            description: The state value for the alarm.
+            returned: always
+            type: str
+        statistic:
+            description: The statistic for the metric associated with the alarm, other than percentile.
+            returned: always
+            type: str
+        threshold:
+            description: The value to compare with the specified statistic.
+            returned: always
+            type: str
+        treat_missing_data:
+            description: Sets how alarm is to handle missing data points.
+            returned: always
+            type: str
+        unit:
+            description: Unit used when storing the metric
+            returned: always
+            type: str
+'''
+
+
+try:
+    import botocore
+except ImportError:
+    pass  # Handled by AnsibleAWSModule
+from ansible_collections.amazon.aws.plugins.module_utils.core import AnsibleAWSModule
+from ansible_collections.amazon.aws.plugins.module_utils.ec2 import AWSRetry
+from ansible.module_utils.common.dict_transformations import camel_dict_to_snake_dict
+
+
+def _describe_alarms(connection, **params):
+    paginator = connection.get_paginator('describe_alarms')
+    return paginator.paginate(**params).build_full_result()
+
+
+def describe_metric_alarms_info(connection, module):
+
+    params = {}
+    params['AlarmNames'] = module.params.get('alarm_names')
+
+    try:
+        describe_metric_alarms_info_response = _describe_alarms(connection, **params)['MetricAlarms']
+    except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
+        module.fail_json_aws(e, msg='Failed to describe cloudwatch metric alarm')
+
+    metric_alarms = []
+    for response_list_item in describe_metric_alarms_info_response:
+        metric_alarms.append(camel_dict_to_snake_dict(response_list_item))
+
+    if len(metric_alarms) == 0:
+        module.exit_json(msg='No metric alarms found for specified options')
```

Comment:
We generally should return an empty list if no alarms are found instead of a message.

Reply:
Fixed.